pkg/services/ngalert/client/client.go

```diff
@@ -70,3 +73,48 @@ func TimeRequest(ctx context.Context, operation string, coll instrument.Collecto
 		coll, toStatusCode, doRequest)
 	return response, err
 }
+
+// TracedClient instruments a request with tracing. It implements Requester.
+type TracedClient struct {
+	client Requester
+	tracer tracing.Tracer
+}
+
+func NewTracedClient(client Requester, tracer tracing.Tracer) *TracedClient {
+	return &TracedClient{
+		client: client,
+		tracer: tracer,
+	}
+}
+
+// Do executes the request.
+func (c TracedClient) Do(r *http.Request) (*http.Response, error) {
```

Comment:
Some inspiration from https://github.com/grafana/grafana/blob/main/pkg/infra/httpclient/httpclientprovider/tracing_middleware.go perhaps using semantic conventions and http trace

For later exercise could potentially use the HTTP client provider instead of your own solution.

Reply:
Good idea - i'll adjust the attributes and naming to be consistent. I picked out a real trace in ops to align these keys on, but the one i picked seems to be different from the infra one

Reply:
Pushed - should be aligned now